File: src/main/java/uss/code/course/dto/response/DepartmentsResponse.java
package uss.code.course.dto.response;

import lombok.Builder;

import java.util.List;

import static lombok.AccessLevel.PRIVATE;

@Builder(access = PRIVATE)
public record DepartmentsResponse(
        List<DepartmentResponse> departmentResponses
) {
    public static DepartmentsResponse of(final List<DepartmentResponse> departmentResponses) {
        return DepartmentsResponse.builder()
                .departmentResponses(departmentResponses)
                .build();
    }
}

File: src/main/kotlin/uss/code/course/dto/response/DepartmentsResponse.kt
package uss.code.course.dto.response

data class DepartmentsResponse(
    val departmentResponses: List<DepartmentResponse>,
) {
    companion object {
        fun of(departmentResponses: List<DepartmentResponse>): DepartmentsResponse {
            return DepartmentsResponse(departmentResponses = departmentResponses)
        }
    }
}
